src/results/state/actions.js

```diff
@@ -16,10 +18,12 @@ export const fetchResultsStarted = () => ({
 });
 
 export const fetchResults = query => (dispatch) => {
-  const queryString = createQueryString(query);
+  const uniprotQueryString = createQueryString(query);
+  const queryUrl = queryString.stringify({
+    query: uniprotQueryString,
+  });
   dispatch(fetchResultsStarted());
-  // TODO we need to use something better to handle url params
-  fetchData(`${apiUrls.advanced_search_terms}/?${queryString}`)
+  fetchData(urljoin(apiUrls.advanced_search_terms, '/?', queryUrl))
```

Comment:
does this want to be `apiUrls .advanced_search`?

Reply:
would url-related activities be better placed in `src/utils/apiUrls.js`? Like having the following function in `apiUrls.js`

```
export const getQueryUrl = uniprotQueryString => urljoin(
  apiUrls.advanced_search_terms,
  '/?',
  queryString.stringify({ query: uniprotQueryString }),
);
```
then in `src/results/state/actions.js` it is called: `fetchData(getQueryUrl(uniprotQueryString))`.